### ISSUE TITLE
Unexpected token error when using backticks for all strings.

### ISSUE THREAD

**Issue by bopfer** [styled-components/stylelint-processor-styled-components#254]
My team is trying to standardize on using backticks for all strings.  We have the eslint rule in place and overall it is working well.  But, we are having one issue with `stylelint` when using typescript types/interfaces and styled-components in the same file.  It seems as though the typescript types are either trying to be parsed as styled-components, or they are just throwing off the the processor in some way.  There are no issues if the types are in a separate file.

## System:
 - OS: macOS 10.14.3
 - CPU: x64 Intel(R) Xeon(R) W-2140B CPU @ 3.20GHz
 - Memory: 482.50 MB / 32.00 GB
 - Shell: 5.0.2 - /usr/local/bin/bash
## Binaries:
 - Node: 11.11.0 - /usr/local/bin/node
 - Yarn: 1.15.2 - /usr/local/bin/yarn
 - npm: 6.7.0 - /usr/local/bin/npm
## npmPackages:
 - styled-components: ^4.1.3 => 4.1.3
 - stylelint: ^9.10.1 => 9.10.1
 - stylelint-config-styled-components: ^0.1.1 => 0.1.1
 - stylelint-processor-styled-components: ^1.5.2 => 1.5.2

## Reproduction

Minimal, contrived repo to show the issue:
https://github.com/bopfer/stylelint-backtick-error

## Steps to reproduce

run `yarn stylelint` on the above repo

## Expected Behavior

No errors

## Actual Behavior

```
yarn run v1.15.2
$ stylelint 'src/**/*.tsx'

src/Message.tsx
 6:11  ✖  Unexpected token (6:11)

error Command failed with exit code 2.
```

**Comment by chinesedfan** [comment]
Looks like babel can't parse this kind of grammar. You can test by https://astexplorer.net/ and file an issue to https://github.com/babel/babel.

> There are no issues if the types are in a separate file.

If there has no styled-components import, the file will be skipped by this processor.

**Comment by bopfer** [comment]
Thanks for the input.  It looks like it does parse properly if I choose `@typescript-eslint/parser`.  See: https://astexplorer.net/#/gist/74619c51d54659abf528e98d767bce9b/a32e2f1181b1bbdbb9d0414e13d06643a9137556

Is it possible to tell this parser to use that somehow?  Sorry if that is a dumb question.  I still need to learn more about ASTs and parsing.  Maybe that totally negates the power of the stylelint processor.

I will search around the Babel project for help as well and create an issue there if needed.

**Comment by chinesedfan** [comment]
@bopfer Sadly this processor is based on babel7. Different parsers may generate different kinds of AST, so we can't change the parser. I will keep this issue open until babel has fixed it.

**Comment by bopfer** [comment]
I asked about it on the Babel Slack and they confirmed that it does appear to be a bug.  I have opened up an issue over there: https://github.com/babel/babel/issues/9706

**Comment by bopfer** [comment]
This has been fixed as of Babel 7.4.3.  It wasn't in the release notes, but the issue's PR was merged to master before the release.

**Comment by chinesedfan** [comment]
Thanks for your feedback.
